File: cpp/0006_ZigZagConversion.cpp
/*
单纯的找规律
*/

class Solution {
public:
    string convert(string s, int numRows) {
       int n=s.size();
       if (numRows<=1) return s;
       int group=(numRows-1)*2;
       string res="";
       for(int i=0;i<numRows;i++)
       {
           for(int j=i;j<n;j=j+group)
           {
               res+=s[j];
               int pos=j+group-i*2;
               if(i!=0 && i!=numRows-1 && pos<n )
               {
                   res+=s[pos];
               }

           }

       }
        return res;

    }
};

File: python/0006_ZigZagConversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1 or numRows >= len(s):
            return s
        res=['']*numRows
        index=0
        step=1

        for n in s:
            res[index]+=n
            if index==0:
                step=1
            if index==numRows-1:
                step=-1
            index += step

        return ''.join(res)
